core/objlist.hpp

```diff
@@ -253,6 +254,14 @@ class ObjList : public ObjListBase {
         DiskStore ds(list_name_);
         BinStream bs;
         bs << objlist_data_;
+        bs << sorted_size_;
+        // TODO(all): Decide whether we need to delete the items
+        // which are marked by del_bitmap_ while writing to the disk
+        // or not
+        // a lazy deletion is needed or not?
```

Comment:
Actually you can use deletion_finalize to get rid of del_bitmap_. For objects that are soft-deleted, no need to save them to disk.